sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.controller.service.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Param,
+  ParseIntPipe,
+  NotFoundException,
+  Injectable,
+} from "@nestjs/common";
+import { StudentRestrictionService } from "../../services";
+import {
+  RestrictionSummaryAPIOutDTO,
+  RestrictionDetailAPIOutDTO,
+} from "./models/restriction.dto";
+import { getUserFullName } from "../../utilities";
+import { ApiNotFoundResponse } from "@nestjs/swagger";
+import { RestrictionNotificationType } from "@sims/sims-db/entities";
+
+@Injectable()
+export class RestrictionControllerService {
+  constructor(
+    private readonly studentRestrictionService: StudentRestrictionService,
+  ) {}
+
+  /**
+   * Get restrictions for a student.
+   * @param studentId id of the student to retrieve restrictions.
+   * @param options query options.
+   * - `filterRestrictions` option to filter restrictions based on notificationType and restrictionCategory.
+   * @returns student restrictions for the provided student id.
+   */
+  async getStudentRestrictions(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    options?: {
+      filterRestrictions?: boolean;
+    },
+  ): Promise<RestrictionSummaryAPIOutDTO[]> {
+    let studentRestrictions =
+      await this.studentRestrictionService.getStudentRestrictionsById(
+        studentId,
+      );
+    if (options?.filterRestrictions) {
+      const filteredStudentRestrictions = studentRestrictions.filter(
+        (studentRestriction) =>
+          studentRestriction.restriction.notificationType !==
+            RestrictionNotificationType.NoEffect &&
+          studentRestriction.restriction.restrictionCategory.toLowerCase() !==
+            "verification",
```

Comment:
Is the check for `verification` defined in some AC or will the ticket be updated?
For instance, the `SINF` (Student submitted application with incorrect SIN expiry date.) is defined with type `Error`, which will make the restrictions be displayed to the student.
This filter will remove `SINF` from the result, is it expected?
@JasonCTang, can you please confirm here also.

Reply:
same with `WTHD ` restriction, students also get an email notification when this restriction is placed on a student, actually, I had created a ticket to review these restrictions and logic https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/1828

Reply:
Filtering out the SINF restriction is NOT expected behaviour; it should be included.  
I believe the filter should just be:
"studentRestriction.restriction.notificationType !== RestrictionNotificationType.NoEffect"
If I am understanding the logic correctly, I believe this means include any restriction that does not have the notification type NoEffect.